### PR TITLE
New package: Etsh

### PR DESCRIPTION
**Pull request by Krush206**

### CODE REVIEW

File: packages/etsh/build.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+TERMUX_PKG_HOMEPAGE=https://etsh.nl
+TERMUX_PKG_DESCRIPTION="An enhanced, backward-compatible port of Thompson Shell"
+TERMUX_PKG_LICENSE="BSD 2-Clause"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=5.4.0
+TERMUX_PKG_SRCURL=https://etsh.nl/src/etsh_5.4.0/etsh-${TERMUX_PKG_VERSION}.tar.xz
```
Comment (xtkoba) [inline packages/etsh/build.sh:6]: "5.4.0" should be substituted by `${TERMUX_PKG_VERSION}`.

File: packages/etsh/build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+TERMUX_PKG_SHA256=fd4351f50acbb34a22306996f33d391369d65a328e3650df75fb3e6ccacc8dce
+TERMUX_BUILD_IN_SRC=true
+
+termux_step_configure() {
+	( cd ${TERMUX_PKG_SRCDIR} ;
+	sh ./mkconfig )
+}
```
Comment (xtkoba) [inline packages/etsh/build.sh:13]: `cd` is not needed here.

File: packages/etsh/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+termux_step_post_configure() {
+	( cd ${TERMUX_PKG_SRCDIR} ;
```
Comment (xtkoba) [inline packages/etsh/build.sh:16]: `cd` is not needed here.

File: packages/etsh/build.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	make &&
+	make install )
```
Comment (xtkoba) [inline packages/etsh/build.sh:18]: I doubt if these are needed here. If necessary redefine `termux_step_make()` and/or `termux_step_make_install()`.

**Review by xtkoba — COMMENTED** [review]
Also fix directory for manpages. Using `$PREFIX/man` raises an error now.

File: packages/etsh/build.sh (hunk 2 above)
Comment (Krush206) [reply packages/etsh/build.sh]: Omitting `cd` results in an error, which is odd. `pwd` returns the correct path, but running `sh` or `make` fails.

File: packages/etsh/build.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+TERMUX_PKG_HOMEPAGE=https://etsh.nl
+TERMUX_PKG_DESCRIPTION="An enhanced, backward-compatible port of Thompson Shell"
+TERMUX_PKG_LICENSE="BSD 2-Clause"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=5.4.0
+TERMUX_PKG_SRCURL=https://etsh.nl/src/etsh_${TERMUX_PKG_VERSION}/etsh-${TERMUX_PKG_VERSION}.tar.xz
+TERMUX_PKG_SHA256=fd4351f50acbb34a22306996f33d391369d65a328e3650df75fb3e6ccacc8dce
+TERMUX_BUILD_IN_SRC=true
```
Comment (xtkoba) [inline packages/etsh/build.sh:8]: The variable name should be `TERMUX_PKG_BUILD_IN_SRC`.

File: packages/etsh/Makefile.patch [hunk 6]
```diff
@@ -0,0 +1,45 @@
+--- a/Makefile	2023-01-09 17:18:03.884999487 -0300
++++ b/Makefile	2023-01-09 23:08:53.080165062 -0300
+@@ -17,14 +17,14 @@
+ # Choose where and how to install the binaries and manual pages.
+ #
+ DESTDIR?=
+-PREFIX?=	/usr/local
++PREFIX?=	/data/data/com.termux/files/usr
```
Comment (xtkoba) [inline packages/etsh/Makefile.patch:8]: Do not hardcode the value `$PREFIX`. Use `@TERMUX_PREFIX@` in a patch.

I doubt if this change is needed here though. We already define `PREFIX` env var.

File: packages/etsh/build.sh (hunk 2 above)
Comment (xtkoba) [reply packages/etsh/build.sh]: See my comment in https://github.com/termux/termux-packages/pull/14430#discussion_r1065283802.

**Pull request merged**